cli/cmd/resource.go

```diff
@@ -31,17 +29,34 @@ func init() {
 }
 
 func runResource(cmd *cobra.Command, args []string) {
-	URL := viper.GetString("zabbix.url")
-	key := viper.GetString("zabbix.api_key")
-	a := zabbix.NewAPI(URL, key)
+	newConfig, err := lib.NewConfigFromFile(lib.ConfigPath)
+	if err != nil {
+		log.Errorf("cannot create config from file. %v", err)
+		return
+	}
+	URL := newConfig.Zabbix.URL
+	key := newConfig.Zabbix.APIKey
+
+	a, err := zabbix.CreateClientUsingAPIKey(URL, key)
+	if err != nil {
+		log.Errorf("cannot find needed params. %v", err)
+		return
+	}
 
 	res, err := a.GetHostsInfo()
 	if err != nil {
-		log.Error("cannot get hosts info.\n", err)
+		log.Errorf("cannot get hosts info. %v", err)
+		return
+	}
+
+	if len(res) == 0 {
+		log.Error("No resource found.")
```

Comment:
should this be an error? if not can we use the appropriate log level

Reply:
Actually it should not. It should just Warn and return out of the function since it is possible Zabbix to not have any servers on it.

Reply:
fixed in updated push.